src/amo/components/Collection/index.js

```diff
@@ -228,6 +237,64 @@ export class CollectionBase extends React.Component<Props> {
     }));
   };
 
+  deleteNote: DeleteAddonNoteFunc = (
+    addonId: number, errorHandler: ErrorHandlerType
+  ) => {
+    const {
+      collection,
+      dispatch,
+      location,
+    } = this.props;
+
+    invariant(addonId, 'addonId is required');
```

Comment:
Good point. I think I was being overzealous with my use of `invariant`. I guess we only need to use it when Flow tells us we need it (because of optional properties).